docs/parser_plugins.md

```diff
@@ -1 +1,423 @@
+In the next section, we convert it.
+
+!!! info "To return or not return"
+    Does the NOMAD base expect an `EntryArchive` object back from `parse`.
+    In NOMAD we use type annotation as much as possible.
+    It also tested in our CI/CD.
+    The type signature of `parse` denotes that it should not return any output, i.e. `-> None`.
+    Instead, the input will be overwritten and later on extracted.
+
+#### Extra: Communicating via Logs
+
+Each entry has an associated log.
+These communicate info or warnings about the processing, as well as debugging info and (critical) errors for tracing bugs.
+Include the latter when contacting us regarding any processing issues.
+
+Here, we use the `logger` object to `info`rm which parser was called.
+
+```python
+...
+from nomad.config import config
+
+configuration = config.get_plugin_entry_point(
+    'nomad_parser_vasp.parsers:xml_entry_point'
+)
+
+class VasprunXMLParser(MatchingParser):
+    def parse(
+        self, mainfile: str, archive: EntryArchive, logger: BoundLogger,
+        child_archives: dict[str, EntryArchive] = None,
+    ) -> None:
+
+    logger.info('VasprunXMLParser.parse', parameter=configuration.parameter)
+    ...
+```
+
+### Instantiating a Schema
+
+<!-- cookie cutter example -->
+
+<!-- contingent instantiation example -->
+
+### Mapping a to Schema
+
+```python
+from structlog.stdlib import BoundLogger
+from nomad.datamodel.datamodel import EntryArchive
+from nomad.config import config
+from nomad.parsing import MatchingParser
+from nomad.parsing.file_parser.mapping_parser import MetainfoParser, XMLParser
+from nomad_parser_vasp.schema_packages.vasp_schema import Simulation
+
+configuration = config.get_plugin_entry_point(
+    'nomad_parser_vasp.parsers:xml_entry_point'
+)
+
+class VasprunXMLParser(MatchingParser):
+    def parse(
+        self, mainfile: str, archive: EntryArchive, logger: BoundLogger,
+        child_archives: dict[str, EntryArchive] = None,
+    ) -> None:
+        logger.info('VasprunXMLParser.parse', parameter=configuration.parameter)
+        data_parser = MetainfoParser(annotation_key='xml', data_object=Simulation())
+        XMLParser(filepath=mainfile).convert(data_parser)
+        archive.data = data_parser.data_object
+```
+
+<!-- TODO Is normalization automatically triggered here? -->
+
+This is all of the code!
+Leveraging `MappingAnnotationModel` clearly makes parsers more concise.
+Breaking down the parser class, `VasprunXMLParser`, it inherits most functionality from `MatchingParser`, including `parse`.
+The `parse` interface is rigid -represents a contract between the NOMAD base and the parser- and is worth studying a bit:
+
+The `MetainfoParser` is linked to the annotated schema. A single schema may contain multiple annotations corresponding to various file formats, e.g. `xml`, `hdf5`, `txt` for VASP. Since we are dealing with computational data, we provide the `Simulation` object as the root of our target.
+
+## From Hierarchies to Schema
```

Comment:
I would consider flipping the order of this section with the last one (I would also combine them into one cohesive section): The section preceding all of this is going to be demonstrating how the manual mapping works. So, it makes sense to start talking first conceptually how you can think about this mapping theoretically in terms of a graph and then demonstrate practically via annotation. Then you say "look how concise the parser code is now!".

Reply:
The graph concept is now introduced at the intro of "From Parser back to NOMAD".
I'll close your comment for now, but we can revisit it later.